src/libpriv/rpmostree-core.cxx

```diff
@@ -1907,10 +1920,43 @@ rpmostree_context_prepare (RpmOstreeContext *self,
               hy_selector_pkg_set (selector, pset);
               if (!hy_goal_install_selector (goal, selector, error))
                 return FALSE;
+
+              map_or (pinned_pkgs_map, dnf_packageset_get_map (pset));
             }
         }
     }
 
+  gboolean we_got_modules = FALSE;
```

Comment:
(Random aside, we can probably use more C++ for stuff like this, i.e. `auto we_got_modules = false;`)

Reply:
Yeah I guess so. Though really the end goal is Rust anyway, so meh... slightly leaning towards sticking with `gboolean` instead to be consistent.